crates/subspace-farmer/src/cluster/controller.rs

```diff
@@ -295,15 +295,28 @@ impl NodeClient for ClusterNodeClient {
     async fn subscribe_slot_info(
         &self,
     ) -> Result<Pin<Box<dyn Stream<Item = SlotInfo> + Send + 'static>>, NodeClientError> {
-        let last_slot_info_instance = Arc::clone(&self.last_slot_info_instance);
         let subscription = self
             .nats_client
             .subscribe_to_broadcasts::<ClusterControllerSlotInfoBroadcast>(None, None)
             .await?
-            .map(move |broadcast| {
-                *last_slot_info_instance.lock() = broadcast.instance;
+            .filter_map({
+                let mut last_slot_number = None;
+                let last_slot_info_instance = Arc::clone(&self.last_slot_info_instance);
 
-                broadcast.slot_info
+                move |broadcast| {
+                    let slot_info = broadcast.slot_info;
+
+                    let maybe_slot_info = if last_slot_number == Some(slot_info.slot_number) {
```

Comment:
I've always been a little confused here. I don't think the later arrival is necessarily the higher one, so shouldn't there be a comparison here? Everywhere else, too.

```rust
let maybe_slot_info = if last_slot_number <= Some(slot_info.slot_number) {
    None
} else { ... }
```